### PR TITLE
Incompatible pointer fixes

### PR DESCRIPTION
**Pull request by chinasaur**
I don't have a good build environment, so all these changes are quick and dirty; I patched the generated .c in my env and confirmed that it fixed compiler warnings, then made equivalent changes in the .pyx without testing. Hope it works out!

### CODE REVIEW

**Comment by chinasaur** [comment]
Yeah, that wasn't a good idea...  I reverted the api_functions.txt changes for now.

**Comment by chinasaur** [comment]
Okay, I think this is ready to pull.  Apologies again for the poor dev strategy.  Next time I have patches I'll be sure to have a reasonable build environment first.

File: api_gen.py [hunk 1]
```diff
@@ -91,10 +91,11 @@ def __init__(self, text):
         self.fname = parts['fname']
         self.sig = parts['sig']
 
-        self.args = self.SIG_PATTERN.findall(self.sig)
+        sig_const_stripped = self.sig.replace('const', '')
+        self.args = self.SIG_PATTERN.findall(sig_const_stripped)
         if self.args is None:
```
Comment (chinasaur) [inline api_gen.py]: I'm not sure None is an adequate check, but out of scope for this change.

Comment (tacaswell) [inline api_gen.py]: `self.args` will only be `None` if the regex fails to match, if it is non-None, then it found one or more matches.

**Comment by andrewcollette** [comment]
Tests pass on Travis and Windows.  Good catch with the rdcc variable too... we'd been using the wrong type for that.

Before I merge, please create a new PR and use rebase to squash things down a bit... right now we have 22 log entries for a relatively minor set of changes.

**Comment by tacaswell** [comment]
For future reference, you don't have to create a new PR, you can just squash and then force-push to the current branch (before you deleted it).

**Comment by andrewcollette** [comment]
@tacaswell... thanks!  Will use this in future.

**Pull request closed without merging**